src/Microsoft.DotNet.Interactive.CSharpProject/Packaging/ProjectAsset.cs

```diff
@@ -84,35 +84,33 @@ private async Task<IAnalyzerResult> LoadResultOrCleanAsync()
         }
     }
 
-    private bool DidPerformCoreCompile(IAnalyzerResult result)
+    private bool DidPerformCoreCompile(BuildDataResults result)
     {
         if (result == null)
         {
             return false;
         }
 
-        var sourceCount = result.SourceFiles?.Length ?? 0;
-        var compilerInputs = result.GetCompileInputs()?.Length ?? 0;
+        var sourceCount = result.BuildProjectData.SourceFiles?.Length ?? 0;
+        // TODO: 
+        var compilerInputs = result.BuildProjectData.CompileInputs?.Length ?? 0;
 
         return compilerInputs > 0 && sourceCount > 0;
     }
 
-    private Task<CodeAnalysis.Workspace> BuildWorkspaceAsync(IAnalyzerResult result)
+    private Task<CodeAnalysis.Workspace> BuildWorkspaceAsync(BuildDataResults result)
     {
-        if (result.TryGetWorkspace(out var ws))
-        {
-            var projectId = ws.CurrentSolution.ProjectIds.FirstOrDefault();
-            var references = result.References;
-            var metadataReferences = references.GetMetadataReferences();
-            var solution = ws.CurrentSolution;
-            solution = solution.WithProjectMetadataReferences(projectId, metadataReferences);
-            ws.TryApplyChanges(solution);
-            return Task.FromResult(ws);
-        }
-        throw new InvalidOperationException("Failed creating workspace");
+        var ws = result.Workspace;
+        var projectId = ws.CurrentSolution.ProjectIds.FirstOrDefault();
+        var references = result.BuildProjectData.References;
+        var metadataReferences = references.GetMetadataReferences();
+        var solution = ws.CurrentSolution;
+        solution = solution.WithProjectMetadataReferences(projectId, metadataReferences);
+        ws.TryApplyChanges(solution);
+        return Task.FromResult(ws);

```

Comment:
It appears that this method was deliberately designed to return a Task because it is passed as an argument to a method that takes a Func<T, Task<U>> nextStep

We want to avoid using Task, then we will have to create another method Then<U>(Func<T, U> nextStep)

We can do this change in another commit.

Reply:
Done. This method is now sync and a new implementation of Then was added